server/neptune/gateway/event/step_generator.go

```diff
@@ -0,0 +1,54 @@
+package event
+
+import (
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/neptune/workflows"
+)
+
+const defaultWorkspace = "default"
+
+type StepGenerator interface {
+	GeneratePlanSteps(repoID string) []workflows.Step
+	GenerateApplySteps(repoID string) []workflows.Step
+}
+
+type TerraformWorkflowStepsGenerator struct {
```

Comment:
Feels a bit premature to have this in a separate struct entirely from the push event handler, especially if all this is doing is just generating the steps.  

We're going to need the project config for figuring out the root definition/directory as well, not just here. I'd just keep this code in the push event handler for now.